var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -226,21 +226,24 @@ class Openmpi(AutotoolsPackage, CudaPackage):
 
     # Additional support options
     variant('atomics', default=False, description='Enable built-in atomics')
-    variant('java', default=False, description='Build Java support')
+    variant('java', default=False, when='@1.7.4:', description='Build Java support')
     variant('static', default=True, description='Build static libraries')
-    variant('sqlite3', default=False, description='Build SQLite3 support')
+    variant('sqlite3', default=False, when='@1.7.3:1', description='Build SQLite3 support')
     variant('vt', default=True, description='Build VampirTrace support')
-    variant('thread_multiple', default=False,
+    variant('thread_multiple', default=False, when='@1.5.4:2',
             description='Enable MPI_THREAD_MULTIPLE support')
-    variant('pmi', default=False, description='Enable PMI support')
-    variant('pmix', default=False, description='Enable PMIx support')
-    variant('wrapper-rpath', default=True,
+    variant('cuda', default=False, when='@1.7:', description='Enable CUDA support')
```

Comment:
I believe there's a bug in spack when overriding unconditional variants from the base class with a conditional variant in the derived class; it seems it's effectively unconditional. Can you turn it into a conflict?

Reply:
Thanks for the info, I've updated accordingly.